torchgeo/trainers/segmentation.py

```diff
@@ -89,20 +89,29 @@ def __init__(self, **kwargs: Any) -> None:
         self.save_hyperparameters()  # type: ignore[operator]
         self.hyperparams = cast(Dict[str, Any], self.hparams)
 
-        self.ignore_zeros = None if kwargs["ignore_zeros"] else 0
-
+        if not (
+            isinstance(kwargs["ignore_index"], int) or (kwargs["ignore_index"] is None)
+        ):
```

Comment:
```suggestion
        if not isinstance(kwargs["ignore_index"], (int, type(None))):
```
Another possible option, no preference on which you choose.